File: examples/node/scrape.mjs
// Scrape a single page with the GhostCrawl Node SDK.
//
//   npm install @ghostcrawl/sdk
//   export GHOSTCRAWL_API_KEY=gck_live_...
//   node scrape.mjs

import { createGhostcrawlClient } from '@ghostcrawl/sdk';

// Reads GHOSTCRAWL_API_KEY from the environment, or pass token: explicitly.
const client = createGhostcrawlClient({ token: process.env.GHOSTCRAWL_API_KEY });

const result = await client.scrape({ url: 'https://example.com', engine: 'chrome', format: 'markdown' });
console.log(result.content ?? result.markdown);

File: examples/python/scrape.py
"""Scrape a single page with the GhostCrawl Python SDK.

    pip install ghostcrawl
    export GHOSTCRAWL_API_KEY=gck_live_...
    python scrape.py

The Python SDK is async — call it from inside an async function.
"""

import asyncio
import os

from ghostcrawl import GhostcrawlClient


async def main() -> None:
    async with GhostcrawlClient(api_key=os.environ["GHOSTCRAWL_API_KEY"]) as client:
        result = await client.scrape(url="https://example.com", engine="chrome")
        print(result["content"])


if __name__ == "__main__":
    asyncio.run(main())
